src/main/scala/org/broadinstitute/dsde/firecloud/EntityClient.scala

```diff
@@ -423,4 +427,68 @@ class EntityClient (requestContext: RequestContext, modelSchema: ModelSchema, go
     }
   }
 
+  def importPFBStatus(workspaceNamespace: String, workspaceName: String, jobId: String, userInfo: UserInfo): Future[PerRequestMessage] = {
+
+    val bucketName = FireCloudConfig.Arrow.bucketName
+
+
+    // list all files in the jobId's subdirectory and strip jobId prefix so we have just the filenames
+    val files = Try(googleServicesDAO.listObjectsAsRawlsSA(bucketName, jobId + "/").map(_.replace(jobId + "/",""))) match {
+      case Success(obs) => obs
+      case Failure(ex) =>
+        logger.warn(s"Error listing GCS objects for prefix gs://$bucketName/$jobId", ex)
+        List("failed to list")
```

Comment:
I suggest keeping the failure here and handing it below. Right now you're creating this sentinel value and then not explicitly accounting for it, leaving it to be interpreted as "a collection of three files that don't look like what we expect", which I'm not the biggest fan of.

Reply:
done!